src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,177 @@
+import java.time.format.DateTimeParseException;
+
+/**
+ * Encapsulates a parser which parses user commands and makes sense of it.
+ *
+ * @author Sean Chin Jun Kai
+ */
+public class Parser {
+    private static String[] args;
+    private static Command action;
+
+    /**
+     * Parses the userInput to retrieve the main command and corresponding arguments.
+     *
+     * @param userInput String that user inputs.
+     * @throws DukeException for invalid arguments.
+     */
+    public static void parseUserResponse(String userInput) throws DukeException {
+        String[] tokens = userInput.split(" ", 2);
+        String command = tokens[0];
+        switch (command) {
+        case "bye": {
+            if (tokens.length != 1) {
+                throw new DukeException("Invalid number of arguments.Only require 1!");
+            }
+            action = Command.BYE;
+            break;
+        }
+        case "list": {
+            if (tokens.length != 1) {
+                throw new DukeException("Invalid number of arguments.Only require 1!");
+            }
+            action = Command.LIST;
+            break;
+        }
+        case "find": {
+            if (tokens.length < 2) {
+                throw new DukeException("Keyword missing!");
+            }
+            action = Command.FIND;
+            args = tokens;
+            break;
+        }
+        case "unmark": {
+            if (tokens.length < 2) {
+                throw new DukeException("Task number missing!");
+            }
+            action = Command.UNMARK;
+            args = tokens;
+            break;
+
+        }
+        case "mark": {
+            if (tokens.length < 2) {
+                throw new DukeException("Task number missing!");
+            }
+            action = Command.MARK;
+            args = tokens;
+            break;
+        }
+        case "delete": {
+            if (tokens.length < 2) {
+                throw new DukeException("Task number missing!");
+            }
+            action = Command.DELETE;
+            args = tokens;
+            break;
+        }
+        case "todo": {
+            if (tokens.length < 2) {
+                throw new DukeException("The description of a todo cannot be empty!");
+            }
+            action = Command.TODO;
+            args = tokens;
+            break;
+        }
+        case "deadline": {
+            if (tokens.length < 2) {
+                throw new DukeException("The description of a deadline cannot be empty!");
+            }
+            action = Command.DEADLINE;
+            args = tokens;
+            break;
+        }
+        case "event": {
+            if (tokens.length < 2) {
+                throw new DukeException("The description of a event cannot be empty!");
+            }
+            action = Command.EVENT;
+            args = tokens;
+            break;
+        }
+        default:
+            throw new DukeException("Invalid command entered!");
```

Comment:
Good cases of exception handling! The exceptions can be more specific if time permits, but since this is a chatbot that focuses heavily on UI, there isn't really a need for too much exception handling either beyond printing error messages.